api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -86,9 +87,9 @@
                                                     (comp form-instance-deleted :id)
                                                     (distinct (keep ::form-instance-changed final)))))
                data-point-deleted (set (keep ::data-point-deleted final))
-               data-point-changed (apply disj (set (keep ::data-point-changed final)) data-point-deleted)
+               data-point-changed (remove #(data-point-deleted (:id %)) (set (keep ::data-point-changed final)))
```

Comment:
Why not remove data-points for surveys that we know have been deleted in this batch? Similar to what we are doing a few lines above (https://github.com/akvo/akvo-flow-api/pull/206/files#diff-0fa2688d1c94f616fb92da2720ce12f4R85) with the form-instances.

Maybe this comment is more general. Do we need to promise that you will not get an update for a child object of an object deleted? 
Because maybe if we promise that, then we also need to remove updates of forms of deleted surveys, form instances of deleted surveys, form instances of deleted data points, ... 
Is this possible? Is it easy? Is 100% accurate?

Reply:
I see that we did "just" the delete-survey removes data-points updates. 

What are your thoughts about the delete-survey should delete form-instances-updates?

Reply:
I just realised that if a survey is deleted, then when we go to load the forms due to a form-instance-update or a form-update, the forms should not be in the Google Datastore anymore. Same goes for a form deleted and a form-instance-update.

This holds true as long as we do not cache the calls to load surveys/forms.

I do not know enough about datapoints.